packages/@aws-cdk/pipelines/lib/codepipeline/codebuild-step.ts

```diff
@@ -198,6 +207,45 @@ export class CodeBuildStep extends ShellStep {
     return this.project.grantPrincipal;
   }
 
+  /**
+   * Additional configuration that can only be configured via BuildSpec
+   *
+   * Contains exported variables
+   *
+   * @default - Contains the exported variables
+   */
+  public get partialBuildSpec(): codebuild.BuildSpec | undefined {
+    this.exportedVarsRendered = true;
+
+    const varsBuildSpec = this.exportedVariables.size > 0 ? codebuild.BuildSpec.fromObject({
+      version: '0.2',
+      env: {
+        'exported-variables': Array.from(this.exportedVariables),
+      },
+    }) : undefined;
+
+    return mergeBuildSpecs(varsBuildSpec, this._partialBuildSpec);
+  }
+
+  /**
+   * Reference a CodePipeline variable defined by the CodeBuildStep.
+   *
+   * The variable must be set in the shell of the CodeBuild step when
+   * it finishes its `post_build` phase.
+   *
+   * @param variableName the name of the variable for reference.
+   */
+  public exportedVariable(variableName: string): string {
+    if (this.exportedVarsRendered && !this.exportedVariables.has(variableName)) {
+      throw new Error('exportVariable(): Pipeline has already been produced, cannot call this function anymore');
+    }
+
+    this.exportedVariables.add(variableName);
+
+    // return `#{${this.variablesNamespace}.${variableName}}`;
+    return new StepOutput(this, variableName).toString();
```

Comment:
I'm confused why we are sending `variableName` as the value for `engineSpecificInformation` to `StepOutput`. It doesn't seem like the variable name is engine specific. Feels like `engineSpecificInformation` could be named better.

Reply:
The idea about "engineSpeicificInformation" is that it is arbitrary information that doesn't matter to the framework.

It matters to the engine, and only the engine gets to interpret it. 

In this case, for the CodePipeline engine, the only thing necessary is the variable name (as the step itself is already known).

I suppose I could make it just a string, and if the engine needs to encode something else into it, it can just JSON-encode something in there...?